specs/stein/approved/amd-sev-libvirt-support.rst

```diff
@@ -0,0 +1,367 @@
+..
+ This work is licensed under a Creative Commons Attribution 3.0 Unported
+ License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+
+====================================================
+libvirt driver launching AMD SEV-encrypted instances
+====================================================
+
+https://blueprints.launchpad.net/nova/+spec/amd-sev
+
+This spec proposes a minimal set of work required in order for nova's
+libvirt driver to support launching of KVM instances which are
+encrypted using `AMD's SEV (Secure Encrypted Virtualization)
+technology <https://developer.amd.com/sev/>`_.
+
+
+Problem description
+===================
+
+Cloud customers are greatly concerned about Spectre, Meltdown, and
+many similar vulnerabilities which enable a malicious VM to harvest
+data from another VM running on the same hardware without being
+detected.  What was once a theoretical attack has now been realized
+with the result that customers are concerned that their data may not
+be adequately protected in a cloud environment.  AMD's SEV offers a
+potential solution for this which transparently encrypts the memory of
+each VM with a unique key.  It can also calculate a signature of the
+memory contents, which can be sent to the guest owner as an
+attestation that the memory was encrypted correctly by the firmware.
+Therefore the solution can also provide some degree of protection
+against a malicious hypervisor or compromised cloud operator, although
+it should probably not be relied upon as the sole line of defense.
+
+Use Cases
+---------
+
+#. Booting an instance with a flavor containing an SEV-specific
+   `required trait extra spec
+   <https://docs.openstack.org/nova/latest/user/flavors.html#extra-specs-required-traits>`_
+   will cause that instance to be defined and booted on a compute host
+   which provides that trait.  Additionally the instance should be
+   created with additional XML in the libvirt domain definition which
+   causes QEMU to boot the instance with SEV enabled.
```

Comment:
as Bryan says, the wording could explain that the traits would trigger two changes in behavior:

* scheduling of the instance on nodes that provide the hardware capability
* defining launch security in the libvirt xml to encrypt the memory of the instance on start

Reply:
Isn't that exactly what it already says?